app/models/neighbour_response.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+class NeighbourResponse
+  include ActiveModel::Model
+
+  STAGES = %w[about_you thoughts response check].freeze
+
+  TAGS = %i[design use privacy light access noise traffic other].freeze
+  RESPONSE_PARAMS  = [:name, :email, :address, :response, :summary_tag, TAGS, :tags, { tags: [] }].freeze
+  PERMITTED_PARAMS = [:stage, :move_next, :move_back, :planning_application_id,
+                      { neighbour_response: RESPONSE_PARAMS }].freeze
+
+  attr_reader :params, :errors
+
+  def initialize(params)
+    @params = params.permit(*PERMITTED_PARAMS)
+    @errors = ActiveModel::Errors.new(self)
+  end
+
+  def stage
+    @stage ||= stage_param.in?(STAGES) ? stage_param : STAGES.first
+  end
+
+  def name
+    response_params[:name].to_s.strip
```

Comment:
If we use the standard Active Model patterns we can change these methods to use setters, e.g.

``` ruby
def name=(value)
  super(value.to_s.strip)
end
```